asllib/doc/ASLTypingReference.tex

```diff
@@ -1618,102 +1797,100 @@ \section{TypingRule.StructuralSubtypeSatisfaction\label{sec:TypingRule.Structura
   \aslto \overname{\Bool}{\vb} \cup \overname{\TTypeError}{\TypeErrorConfig}
 \]
 tests whether a type $\vt$ \emph{structurally-subtype-satisfies} a type $\vs$ is environment $\tenv$,
-returning a type error if one is detected.
+returning the result $\vb$ or a type error, if one is detected.
 
 \subsection{Prose}
-$\vt$ structurally-subtype-satisfies $\vs$ in environment \tenv, if one of the following applies:
+One of the following applies:
+\begin{itemize}
+\item All of the following apply (\textsc{error1}):
   \begin{itemize}
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of an integer type;
-    \item $\vt$ has the underlying type of an integer type.
-    \end{itemize}
-
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of an integer type;
-    \item $\vt$ has the underlying type of an integer type.
-    \end{itemize}
-
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of a string type;
-    \item $\vt$ has the underlying type of a string type.
-    \end{itemize}
+  \item obtaining the \underlyingtype\ of $\vt$ gives a type error;
+  \item the rule results in a type error.
+  \end{itemize}
 
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of a boolean type;
-    \item $\vt$ has the underlying type of a boolean type.
+\item All of the following apply (\textsc{error2}):
+  \begin{itemize}
+    \item obtaining the \underlyingtype\ of $\vt$ gives a type $\vttwo$;
+    \item obtaining the \underlyingtype\ of $\vs$ gives a type error;
+    \item the rule results in a type error.
     \end{itemize}
 
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of an enumeration type;
-    \item $\vt$ has the underlying type of an enumeration type;
-    \item $\vs$ and $\vt$ have the same enumeration literals.
-   \end{itemize}
+\item All of the following apply (\textsc{different\_labels}):
+  \begin{itemize}
+  \item the underlying types of $\vt$ and $\vs$ have different labels,
+  which means that they both have totally different kinds of types (for example, \texttt{integer} and \texttt{real}),
+  which is determined by comparing their AST labels;
+  \item $\vb$ is $\False$.
+  \end{itemize}
 
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of a bitvector type with determined width $\vw$;
-    \item One of the following applies:
-      \begin{itemize}
-      \item $\vt$ has the underlying type of a bitvector type of determined width $\vw$;
-      \item $\vt$ has the underlying type of a bitvector type of undetermined width.
-      \end{itemize}
-    \end{itemize}
+\item All of the following apply (\textsc{simple}):
```

Comment:
I am not sure what the mention in parentheses is - does this correspond to something in the code?

Reply:
When a rule splits into cases, I'm using the case names in the Prose paragraph to make it easy to relate the Prose cases and the rules in the Formally paragraphs.

Reply:
I added an explanation.

Reply:
> When a rule splits into cases, I'm using the case names in the Prose paragraph to make it easy to relate the Prose cases and the rules in the Formally paragraphs.

please would you be able to make that explicit somewhere? maybe in the reading preamble?